testsuite/MDAnalysisTests/core/test_requires.py

```diff
@@ -43,3 +43,20 @@ def mass_multiplier(ag1, ag2, scalar):
         result = mass_multiplier(u.atoms[:10], u.atoms[20:30], 4.0)
 
         assert_(isinstance(result, np.ndarray))
+
+    def test_failure_errormessage(self):
+        # failures should list all required attributes not
+        # just the first one
+        @requires('cats', 'dogs', 'frogs')
```

Comment:
Yeah that always confuses me, but if you write it out as a nested function it makes sense...

```python
requires('cats')(my_func)

# Is equivalent to:
cat_requires = requires('cats')
@cat_requires
def my_func():
   etc
```
So `requires('cats')` creates the decorator, which is then applied to `my_func`